tests/5.1/target_update/test_target_update_iterator.c

```diff
@@ -0,0 +1,76 @@
+//===--- test_target_update_iterator.c --------------------------------------------------------------===//
+//
+//  OpenMP API Version 5.1 Aug 2021
+//
+//  This test usese the target update directive with the iterator clause
+//  to update the iterative values.
+//
+//  This example has been adapted from the 5.2 OpenMP Examples document,
+//  "Multidependences Using Iterators" and "Simple target data and target update
+//  constructs"
+//
+////===--------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+typedef struct test_struct{
+    size_t len;
+    int *data;
+} test_struct_t;
+
+void init(struct test_struct *s ){
+    for(size_t i = 0; i < s->len; i++){
+        s->data[i] = i;
+    }
+}
+
+void init_again(struct test_struct *s ){
+    for(size_t i = 0; i < s->len; i++){
+        s->data[i] = i+1;
+    }
+}
+
+int test_target_update_iterator() {
+    int errors = 0;
+    int A[N]; // array for testing
+
+    for(int i = 0; i < N; i++){
+        A[i] = 0;
+    }
+    test_struct_t new_struct;
+
+    init(&new_struct);
+    #pragma omp target map(to: new_struct) map(from: A[:N])
+    {
+        #pragma omp parallel for
+        for(int i = 0; i < N; i++){
+            A[i] = new_struct.data[i];
+        }
+    }
+    init_again(&new_struct);
+    // update with new values, i+i+1
+    #pragma omp target update to(iterator(it = 0:N), new_struct.data[it])
```

Comment:
I think it should be:
```suggestion
    #pragma omp target update to(iterator(it = 0:N): new_struct.data[it])
```
_Disclaimer: I have not fully read the example and I don't have (tried) a compiler support iterators in map._

Reply:
I believe the comma is correct, spec defines as:

> where iterators-definition is one of the following:
> `iterator-specifier [, iterators-definition ]`